### PR TITLE
Programmed actions: Automatically disable and log when used

### PR DESCRIPTION
**Pull request by jenf**
* Move to a player_log for things that should only be seen by a player (not yet filtered to the player)
* Automatically remove programmed actions on round change

### CODE REVIEW

File: lib/engine/step/buy_sell_par_shares.rb [hunk 1]
```diff
@@ -281,7 +281,9 @@ def activate_program_buy_shares(entity, program)
         end
         share = corporation.ipo_shares.first
         if can_buy?(entity, share.to_bundle)
-          [Action::BuyShares.new(entity, shares: share)]
+          action = Action::BuyShares.new(entity, shares: share)
+          action.user = 'programmed'
```
Comment (tobymao) [inline lib/engine/step/buy_sell_par_shares.rb:285]: can we do this another way?
like programmed = true

Comment (jenf) [reply lib/engine/step/buy_sell_par_shares.rb:285]: Sure, do you want the JSON to change then?

Comment (tobymao) [reply lib/engine/step/buy_sell_par_shares.rb:285]: i wonder if this is necessary at all. i think people might not want it to be any different from a normal action. thoughts on just having it appear as an instant user action?

Comment (jenf) [reply lib/engine/step/buy_sell_par_shares.rb:285]: It feels like it might confuse people since it's instant

Comment (tobymao) [reply lib/engine/step/buy_sell_par_shares.rb:285]: i think it would be fine, thoughts about just trying that for now until people complain?

Comment (jenf) [reply lib/engine/step/buy_sell_par_shares.rb:285]: fair enough, will remove until someone complains

**Review by tobymao — APPROVED** [review]

**Pull request merged**